test/e2e/appcr_svcaccnt_delete_order_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestAppCRSvcAccntDeleteOrder(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	yaml := `
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default-ns-sa
+  namespace: kapp-test
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: default-ns-role
+  namespace: kapp-test
+rules:
+- apiGroups: ["*"]
+  resources: ["*"]
+  verbs: ["*"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: default-ns-role-binding
+  namespace: kapp-test
+subjects:
+- kind: ServiceAccount
+  name: default-ns-sa
+  namespace: kapp-test
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: default-ns-role
+---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: simple-app-cr
+  namespace: kapp-test
+spec:
+  serviceAccountName: default-ns-sa
+  fetch:
+  - git:
+      url: https://github.com/k14s/k8s-simple-app-example
+      ref: origin/develop
+      subPath: config-step-2-template
+  template:
+  - ytt: {}
+  deploy:
+  - kapp: {}
+`
+	name := "test-appcr-svcaccnt-delete-order"
+	kcappname := "kc"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", kcappname})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("install kapp-controller on cluster", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml", "-a", kcappname},
+			RunOpts{})
+		// TODO: how to test presence of kc
```

Comment:
this seems like a pretty heavy dependency to take in e2e tests. i think we should be able to add a lower level test that focuses specifically on ordering -- similar to https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/diffgraph/change_graph_cf_for_k8s_test.go#L18-L46

Reply:
Thanks for suggestion, I was looking into the same. Will update the changes soon.